tests/test_kb_dataset.py

```diff
@@ -0,0 +1,310 @@
+import argparse
+import sys
+import unittest
+from collections import defaultdict
+from difflib import ndiff
+from pathlib import Path
+from pprint import pformat
+from typing import Iterator, Optional, Union
+
+sys.path.append(str(Path(__file__).parent.parent))
+
+import datasets
+
+from schemas.kb import features
+
+OFFSET_ERROR_MSG = (
+    "\n"
+    "There are features with wrong offsets!"
+    "This is not a hard failure, as it is common for this type of datasets"
+    "However, if the error list is long (e.g. >10) you should double check your code"
+)
+
+
+def _get_example_text(example: dict) -> str:
+    """
+    Get full text example
+    """
+    return " ".join([t for p in example["passages"] for t in p["text"]])
+
+
+def _check_offsets(
+    example_text: str,
+    offsets: list[list[int]],
+    texts: list[str],
+) -> Iterator:
+
+    for idx, (start, end) in enumerate(offsets):
+
+        try:
+            text = texts[idx]
+
+        # TODO: riase error?
+        except IndexError:
+            print("Pairing of a list of offsets with a single text is not allowed!")
+
+        by_offset_text = example_text[start:end]
+
+        if by_offset_text != text:
+            yield f" text:`{text}` != offsets_text:`{by_offset_text}`"
+
+
+class TestKBDataset(unittest.TestCase):
+
+    PATH: str
+    DATA_DIR: str
+    USE_AUTH_TOKEN: Optional[Union[bool, str]]
+
+    def setUp(self) -> None:
+
+        self.dataset_source = datasets.load_dataset(
+            self.PATH,
+            name="source",
+            data_dir=self.DATA_DIR,
+            use_auth_token=self.USE_AUTH_TOKEN,
+        )
+
+        self.dataset_bigbio = datasets.load_dataset(
+            self.PATH,
+            name="bigbio",
```

Comment:
probably want to allow a little bit of flexibility here ... every once in a while we'll have one dataset that supports multiple tasks and so might have  `bigbio-translation` and `bigbio-kb` in one dataset.

Reply:
I'll add an argument for that!